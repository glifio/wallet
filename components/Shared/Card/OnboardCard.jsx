import React, { forwardRef } from 'react'
import Box from '../Box'

const OnboardCard = forwardRef(({ ...props }, ref) => (
  <Box
    display='inline-block'
    width='100%'
    maxWidth={13}
    minHeight={10}
    p={3}
    border={1}
    borderRadius={2}
    borderWidth={1}
    overflow='hidden'
    ref={ref}
    {...props}
  />
))

export default OnboardCard
