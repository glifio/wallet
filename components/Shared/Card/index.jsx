import React, { forwardRef } from 'react'
import Box from '../Box'

const Card = forwardRef(({ ...props }, ref) => (
  <Box
    display='inline-block'
    width={300}
    p={3}
    border={1}
    borderRadius={2}
    borderWidth={1}
    color='text'
    overflow='hidden'
    ref={ref}
    {...props}
  />
))

export default Card
