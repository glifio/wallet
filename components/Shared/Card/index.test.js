import { cleanup, render, act, fireEvent } from '@testing-library/react'
import Card from '.'

describe('Card', () => {
  afterEach(cleanup)
  test('renders the card', () => {
    const { container } = render(<Card />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders the card with the right attributes', () => {
    const { container } = render(
      <Card
        position='flex'
        flexDirection='row'
        border='1'
        width='auto'
        alignItems='center'
        justifyContent='space-between'
        bg='blue'
        p={4}
      />
    )

    const card = container.querySelector('div')
    expect(card).toHaveStyle('width: auto;')
    expect(card).toHaveStyle('position: flex;')
    expect(card).toHaveStyle('align-items: center;')
    expect(card).toHaveStyle('justify-content: space-between;')
    expect(card).toHaveStyle('background-color: blue;')
    expect(card).toHaveStyle('padding: 32px;')
  })

  test('clicking Card calls onClick', () => {
    const mockOnClick = jest.fn()
    const { container } = render(
      <Card
        position='flex'
        onClick={mockOnClick}
        border='1'
        width='auto'
        alignItems='center'
        justifyContent='space-between'
        bg='blue'
        p={4}
      />
    )

    act(() => {
      fireEvent.click(container.querySelector('div'))
    })

    expect(mockOnClick).toHaveBeenCalled()
  })
})
