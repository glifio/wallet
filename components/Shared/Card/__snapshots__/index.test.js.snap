// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders the card 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  border: 1px;
  border-radius: 2px;
  border-width: 1px;
  padding: 16px;
  display: inline-block;
  width: 300px;
  overflow: hidden;
}

<div
  class="c0"
  display="inline-block"
  overflow="hidden"
  width="300"
/>
`;
